[WIP] feat : Added Event Streams DA (Fully configurable variations)

Khuzaima05 opened the pull request:
### Description

Added Event Streams DA (Fully configurable variations) 

issue: https://github.ibm.com/GoldenEye/issues/issues/13014

### Release required?
<!--- Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning). --->

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

<!--- If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module. --->

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

Khuzaima05 commented:
/run pipeline

Khuzaima05 commented:
/run pipeline

Khuzaima05 commented:
The upgrade test is failing because it clones the main branch and looks for the standard DA repo, which doesn’t exist in main yet—we’re introducing it in our current branch. The main branch only contains the quickstart and enterprise DAs, which we’ve are updating. Hence, we are skipping the upgrade test.

<img width="1340" alt="Screenshot 2025-05-14 at 2 16 17 PM" src="https://github.com/user-attachments/assets/ccc8c226-2e2a-4212-a4bf-12e464d15946" />

Khuzaima05 commented:
/run pipeline

Khuzaima05 commented:
/run pipeline

Khuzaima05 commented:
/run pipeline

Khuzaima05 commented:
/run pipeline

Khuzaima05 commented:
/run pipeline

Khuzaima05 commented:
/run pipeline

Khuzaima05 commented:
/run pipeline

Khuzaima05 commented:
/run pipeline

Khuzaima05 commented:
/run pipeline

Ak-sky commented on solutions/quickstart/DA-prefix.md:
Not sure if we needed this doc file as we are already linking this [doc URL](https://terraform-ibm-modules.github.io/documentation/#/da-implementation-guidelines.md) to the prefix variable.

Khuzaima05 replied:
We don't need now , I have removed that.

Ak-sky commented on solutions/quickstart/DA-schemas-topics.md:
Not needed here as the same information is also available in security-enforced DA docs, just point it to the variable description.

Khuzaima05 replied:
Done

Ak-sky commented on solutions/quickstart/variables.tf:
```diff
@@ -55,37 +71,53 @@ variable "plan" {
     }
   ))
-  description = "The list of topics to apply to resources. Only one topic is allowed for Lite plan instances. [Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/tree/main/solutions/quickstart/DA-schemas-topics-cbr.md)."
+  description = "The list of topics to apply to resources. [Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/tree/main/solutions/enterprise/DA-schemas-topics-cbr.md#options-with-topics)."
```
```suggestion
  description = "The list of topics to apply to resources. [Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/tree/main/solutions/security-enforced/DA-schemas-topics-cbr.md#options-with-topics)."
```

Ak-sky commented on solutions/quickstart/variables.tf:
```diff
@@ -55,37 +71,53 @@ variable "plan" {
+
 variable "service_credential_names" {
-  description = "The mapping of names and roles for service credentials that you want to create for the Event streams.[Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/tree/main/solutions/quickstart/DA-types.md#svc-credential-name)"
+  description = "The mapping of names and roles for service credentials that you want to create for the Event streams.[Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/tree/main/solutions/enterprise/DA-types.md#svc-credential-name)"
```
```suggestion
  description = "The mapping of names and roles for service credentials that you want to create for the Event streams.[Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/tree/main/solutions/security-enforced/DA-types.md#svc-credential-name)"
```

Ak-sky commented on solutions/quickstart/variables.tf:
```diff
@@ -122,9 +154,9 @@ variable "service_credential_secrets" {
   default     = []
-  description = "Service credential secrets configuration for Event Streams. [Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/tree/main/solutions/quickstart/DA-types.md#service-credential-secrets)."
+  nullable    = false
+  description = "Service credential secrets configuration for Event Streams. [Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/tree/main/solutions/enterprise/DA-types.md#service-credential-secrets)."
```
```suggestion
  description = "Service credential secrets configuration for Event Streams. [Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/tree/main/solutions/security-enforced/DA-types.md#service-credential-secrets)."
```

Ak-sky reviewed (changes requested):
- Left few comments, also need to update the URL in variables in exisitng code to use `security-enforced` instead of `enterprise`.
- Few typos below needs to be fixed
  ```
  solutions/security-enforced/README.md:5: wan't ==> want, wasn't
  solutions/security-enforced/README.md:12: instace ==> instance
  solutions/security-enforced/README.md:57: Steams ==> Streams
  solutions/security-enforced/README.md:58: Steams ==> Streams
  solutions/security-enforced/main.tf:169: intance ==> instance, intense
  solutions/security-enforced/variables.tf:66: Steams ==> Streams
  solutions/security-enforced/variables.tf:72: Steams ==> Streams
  solutions/quickstart/README.md:45: Steams ==> Streams
  solutions/quickstart/README.md:46: Steams ==> Streams
  solutions/quickstart/main.tf:37: intance ==> instance, intense
  solutions/quickstart/variables.tf:76: Steams ==> Streams
  solutions/quickstart/variables.tf:82: Steams ==> Streams
  ```

ocofaigh reviewed (changes requested):
- remove `support_ibm` tag - its not IBM supported yet, its only in community
- add the following support statement: 
`"support_details": "This product is in the community registry, as such support is handled through the originated repo. If you experience issues please open an issue in the repository [https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/issues](https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/issues). Please note this product is not supported via the IBM Cloud Support Center.",`

Khuzaima05 commented on solutions/security-enforced/README.md:
```diff
@@ -0,0 +1,105 @@
+- Topics to apply to resources.
+- Schemas to apply to resources.
+- Mirroring of existing event stream instace.
+
```
Done

Khuzaima05 commented:
> * Left few comments, also need to update the URL in variables in existing code to use `security-enforced` instead of `enterprise`.
> * Few typos below needs to be fixed, [PR](https://github.com/terraform-ibm-modules/common-dev-assets/pull/1106) in progress to add codespell to check for typos.
> 
> ```
> modules/fscloud/variables.tf:10: Steams ==> Streams
> modules/fscloud/variables.tf:16: Steams ==> Streams
> solutions/quickstart/README.md:45: Steams ==> Streams
> solutions/quickstart/README.md:46: Steams ==> Streams
> examples/basic/variables.tf:27: Steams ==> Streams
> examples/fscloud/variables.tf:27: Steams ==> Streams
> modules/fscloud/README.md:29: Steams ==> Streams
> modules/fscloud/README.md:47: Steams ==> Streams
> solutions/quickstart/main.tf:37: intance ==> instance, intense
> README.md:164: Steams ==> Streams
> solutions/security-enforced/variables.tf:66: Steams ==> Streams
> solutions/security-enforced/variables.tf:72: Steams ==> Streams
> solutions/security-enforced/README.md:5: wan't ==> want, wasn't
> solutions/security-enforced/README.md:12: instace ==> instance
> solutions/security-enforced/README.md:57: Steams ==> Streams
> solutions/security-enforced/README.md:58: Steams ==> Streams
> variables.tf:27: Steams ==> Streams
> examples/complete/variables.tf:27: Steams ==> Streams
> examples/complete/variables.tf:33: Steams ==> Streams
> solutions/security-enforced/main.tf:169: intance ==> instance, intense
> solutions/quickstart/variables.tf:76: Steams ==> Streams
> solutions/quickstart/variables.tf:82: Steams ==> Streams
> ```

Done

Khuzaima05 commented:
/run pipeline

Khuzaima05 commented:
/run pipeline

Khuzaima05 commented on ibm_catalog.json:
Done, I have added observability add-ons as per mentioned [here](https://github.ibm.com/GoldenEye/issues/issues/13599#:~:text=Any%20DA%20that,this%20PR).

Khuzaima05 commented on solutions/quickstart/variables.tf:
@Ak-sky , I have not added CBR in quickstart variation as it was not already there in existing quickstart variation
ref: https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/blob/0a67c7dbf97809dfa112f3eb393dcdc18dac6d05/solutions/quickstart/main.tf#L18

Khuzaima05 commented:
/run pipeline

Ak-sky commented:
@ocofaigh, could you please give a final look at this PR.

Khuzaima05 commented:
Tested it successfully deploying with all add-ons in project: 

<img width="1465" alt="Screenshot 2025-06-17 at 2 49 41 PM" src="https://github.com/user-attachments/assets/c252c3e7-671d-42a1-b61c-e4f2a9b94f9d" />

Khuzaima05 commented:
/run pipeline

Ak-sky reviewed (dismissed):
LGTM

ocofaigh reviewed (changes requested):
We are not consistent with our inputs. some are named `event_stream_` and some `event_streams_`. Since this is a single service DA, I don’t know why you are adding this prefix to all the vars? It only deploys 1 service instance

ocofaigh commented on solutions/quickstart/README.md:
```diff
@@ -1,11 +1,78 @@
+#  Cloud automation for Event Streams (quickstart)
 
-This architecture creates an instance of IBM Event Streams for IBM Cloud.
+This solution supports provisioning and configuring the following infrastructure:
```
See https://github.ibm.com/GoldenEye/issues/issues/13599 - point 2 for guidance on what should be in a DA readme

Khuzaima05 replied:
updated

ocofaigh commented on ibm_catalog.json:
```diff
@@ -9,71 +9,129 @@
           "install_type": "fullstack",
           "working_directory": "solutions/quickstart",
-          "configuration": [
+          "compliance": {
```
quickstart would not claim any compliance controls

ocofaigh commented on ibm_catalog.json:
```diff
@@ -9,71 +9,129 @@
+          },
+          "architecture": {
+            "descriptions": "This architecture supports provisioning and configuring quickstart Event Streams.",
+            "features": [
```
This does not comply with the guidance in https://github.ibm.com/GoldenEye/issues/issues/13599 - see bullet point 8 (we might have to tweek the text for this DA since it has a quickstart)

Khuzaima05 replied:
updated

ocofaigh commented on ibm_catalog.json:
```diff
@@ -106,30 +176,20 @@
-            },
-            {
-              "key": "access_tags"
+              "key": "existing_resource_group_name",
```
missing the display name: `"display_name": "resource_group",`

Khuzaima05 replied:
updated

ocofaigh commented on ibm_catalog.json:
```diff
@@ -272,28 +511,35 @@
+            },
+            {
+              "key": "existing_resource_group_name",
+              "required": true,
```
should not be force required

Khuzaima05 replied:
Done

Khuzaima05 commented on ibm_catalog.json:
```diff
@@ -106,30 +176,20 @@
-            {
-              "key": "access_tags"
+              "key": "existing_resource_group_name",
+              "required": true,
```
Done

Khuzaima05 commented on solutions/security-enforced/DA-schemas-topics-cbr.md:
```diff
@@ -85,8 +85,8 @@ The following example includes all the configuration options for cbr rule.
-                    "name" : "endpointType",
-                    "value" : "private"
+                    name : "endpointType",
+                    value : "private"
```
updated

Khuzaima05 commented on solutions/security-enforced/README.md:
```diff
@@ -0,0 +1,107 @@
+#  Cloud automation for Event Streams (Security-Enforced)
+
```
updated

Khuzaima05 commented:
> We are not consistent with our inputs. some are named `event_stream_` and some `event_streams_`. Since this is a single service DA, I don’t know why you are adding this prefix to all the vars? It only deploys 1 service instance

updated

Khuzaima05 commented:
Updated Tile:


<img width="1686" alt="Screenshot 2025-06-19 at 2 20 06 AM" src="https://github.com/user-attachments/assets/c92f19fd-89f7-4802-842b-e56badf36661" />

Khuzaima05 commented:
/run pipeline

Khuzaima05 commented:
/run pipeline

ocofaigh reviewed (changes requested):
- The diagram is missing Secrets Manager? Both variations support creating sericce credentials and storing in secrets manager. And the addons config is also missing secrets manager
- I don't think quickstart should have addons enabled. Its not really "quickstart" then

Khuzaima05 commented:
> * The diagram is missing Secrets Manager? Both variations support creating service credentials and storing in secrets manager. And the addons config is also missing secrets manager
> * I don't think quickstart should have addons enabled. Its not really "quickstart" then

Added secrets manager as add-ons and updated the reference diagram.

For quickstart add-ons, I have made all add-ons  `off-by-default`.

Khuzaima05 commented:
/run pipeline

ocofaigh commented on solutions/security-enforced/README.md:
```diff
@@ -0,0 +1,3 @@
+#  Cloud automation for Event Streams (Security-Enforced)
```
Security-enforced
